pallets/asset-index/src/lib.rs

```diff
@@ -281,6 +302,30 @@ pub mod pallet {
             Ok(().into())
         }
 
+        /// Registers a new asset in the index together with its availability
+        ///
+        /// Only callable by the admin origin and for assets that are not yet registered.
+        #[pallet::weight(10_000)] // TODO: Set weights
+        pub fn register_asset(
+            origin: OriginFor<T>,
+            asset_id: T::AssetId,
+            availability: AssetAvailability,
+        ) -> DispatchResult {
+            T::AdminOrigin::ensure_origin(origin.clone())?;
+
+            Assets::<T>::try_mutate(asset_id, |maybe_available| -> DispatchResult {
+                // allow new assets only
+                ensure!(
+                    maybe_available.replace(availability.clone()).is_none(),
```

Comment:
If asset exists, we should roll back the storage

Reply:
`try_mutate` only mutates if `Ok` is returned.